### PR TITLE
Expose resolved services as a private API

### PR DESCRIPTION
**Pull request by shannonmoeller**
RFC: https://github.com/fusionjs/rfcs/pull/11
Depended on by: https://github.com/fusionjs/fusion-react/pull/163

### CODE REVIEW

File: src/base-app.js [hunk 1]
```diff
@@ -73,7 +75,12 @@ class FusionApp {
       aliases: new Map(),
       enhancers: [],
     };
-    this.registered.set(getTokenRef(token), {value, aliases, enhancers, token});
+    this.registered.set(getTokenRef(token), {
+      value,
+      aliases,
+      enhancers,
+      token,
+    });
```
Comment (shannonmoeller) [inline src/base-app.js]: This formatting was changed by prettier.

File: src/base-app.js [hunk 2]
```diff
@@ -101,7 +108,12 @@ class FusionApp {
     if (enhancers && Array.isArray(enhancers)) {
       enhancers.push(enhancer);
     }
-    this.registered.set(getTokenRef(token), {value, aliases, enhancers, token});
+    this.registered.set(getTokenRef(token), {
+      value,
+      aliases,
+      enhancers,
+      token,
+    });
```
Comment (shannonmoeller) [inline src/base-app.js]: This formatting was changed by prettier.

**Comment by rtsao** [comment]
This effectively makes all services enumerable, right? I am unsure if this a good thing or bad thing or neutral. Would this compromise any guarantees with regard to the DI lifecycle?

Perhaps a more conservative API change would only expose a service retrieval function (taking the token as an argument). I think this would still satisfy the desired use case.

**Comment by shannonmoeller** [comment]
I’d love to hide the map if possible, but how would you implement the service retrieval method without storing the services map as a property of the class as this PR is already doing?

**Comment by shannonmoeller** [comment]
@rtsao Just thought of the pattern for this. Will update the PR.

**Comment by ganemone** [comment]
!merge

**Pull request merged**